### ISSUE TITLE
Failure cases in avatar data

### ISSUE THREAD

**Issue by Yifehuang97** [aipixel/GPS-Gaussian#32]
Hi, thanks for your work! When I am using the default setting on avatar, I saw some failure cases like:

![Fail2](https://github.com/aipixel/GPS-Gaussian/assets/89707220/80ff2e52-3d73-4773-bac2-896d1dca19cb)
![Fail1](https://github.com/aipixel/GPS-Gaussian/assets/89707220/1445dc34-51ed-4c75-b983-1ac86d041f1b)

It seems the Gaussian regression network fails to learn how to combine the Gaussian from the left/right views. Also, the stage 2 loss seems didn't converge well.
![image](https://github.com/aipixel/GPS-Gaussian/assets/89707220/36fc4f3a-42cc-45d9-ad6e-98e5eb9cbd7c)

I also am not sure whether the stage 1 result is good enough; the final validation EPE is around 1.767.
![image](https://github.com/aipixel/GPS-Gaussian/assets/89707220/20fd02fd-491f-4f14-924d-6efe0a073ece)

Thanks for your time!

**Comment by ShunyuanZheng** [comment]
I think the depth estimation in Stage 1 is well-trained since an EPE around 1.767 is reasonable. However, the failure case in fig2 is confusing. The obvious mismatch of two partial Gaussian points is witnessed, and I think it is mainly caused by camera parameters. I suggest saving the point clouds of both views either in Stage 1 or Stage 2.

**Comment by Yifehuang97** [comment]
Thanks for your reply!

This is the EPE metric of validation set in stage 2:
![image](https://github.com/aipixel/GPS-Gaussian/assets/89707220/60f05443-5fde-4dd8-852e-93de5cb699c5)

For saving the point clouds, do you mean to call the depth2pc to get the 3D position to see if the result of left/right is consistent?

Thank you for your help!

**Comment by ShunyuanZheng** [comment]
The EPE also seems reasonable : )

**Comment by Yifehuang97** [comment]
Thank you so much!

**Comment by Yifehuang97** [comment]
I apologize for the inconvenience of reaching out once more. I've resolved a bug in my previous code, which now produces reasonable results. However, it appears that both the L1 loss and SSIM loss are not converging as expected. Furthermore, the quality of the rendered images does not seem to improve with additional training iterations. Could you suggest any potential reasons for this issue?

Results from the first iteration on the **training set**:
![image](https://github.com/aipixel/GPS-Gaussian/assets/89707220/764f87d5-93b9-4d2d-9be8-263f1a8eeef0)

Results after 100,000 iterations on the **training set:**
![image](https://github.com/aipixel/GPS-Gaussian/assets/89707220/d25ebf13-6167-4d61-a0a9-61ecaa47a232)

Here is the training loss:
![image](https://github.com/aipixel/GPS-Gaussian/assets/89707220/58a635d0-0b3b-4a6b-aa98-ceb960313233)

Thank you so much, and sorry for the inconvenience.

**Comment by ShunyuanZheng** [comment]
Sorry for the late reply. I got a similar result when I trained under a half-body setup where the severe self-occlusion caused many holes in novel views. The large-scale Gaussians are predicted to compensate for the missing areas. However, it should not be the reason for the head NVS since there is not any occlusion. I think you can manually set the scale of Gaussians to zero in the Gaussian rasterization and compare the rendered novel view image and the ground truth. I think there still exists a mismatch between them which is potentially caused by the incorrect camera parameters.

**Comment by Yifehuang97** [comment]
Thank you!
